### ISSUE TITLE
RangeError when adding path with windows_path

### ISSUE THREAD

**Issue by TeroPihlaja** [chef/chef#11349]
## Description

RangeError when adding path with windows_path

```
RangeError: bignum too big to convert into `long'
```

## Chef Version
16.13.16

## Platform Version
Windows Server 2019

## Replication Case
I think it happens on all calls to windows_path, for example:
```
windows_path "C:/Program Files/PostgreSQL/13/bin/" do
  action :add
end
```

## Client Output
```
    * windows_env[path] action modify

      ================================================================================
      Error executing action `modify` on resource 'windows_env[path]'
      ================================================================================

      RangeError
      ----------
      bignum too big to convert into `long'

      Resource Declaration:
      ---------------------
      # In C:/cinc-project/cinc/embedded/lib/ruby/gems/2.7.0/gems/chef-16.13.16-universal-mingw32/lib/chef/resource/windows_path.rb

       72:         env "path" do
       73:           action :modify
       74:           delim ::File::PATH_SEPARATOR
       75:           value path.tr("/", '\\')
       76:         end
       77:       end

      Compiled Resource:
      ------------------
      # Declared in C:/cinc-project/cinc/embedded/lib/ruby/gems/2.7.0/gems/chef-16.13.16-universal-mingw32/lib/chef/resource/windows_path.rb:72:in `block in <class:WindowsPath>'

      windows_env("path") do
        action [:modify]
        default_guard_interpreter :default
        declared_type :env
        cookbook_name "..."
        value "C:\\Program Files\\PostgreSQL\\13\\bin\\;%SystemRoot%\\system32;%SystemRoot%;%SystemRoot%\\System32\\Wbem;%SYSTEMROOT%\\System32\\WindowsPowerShell\\v1.0\\;%SYSTEMROOT%\\System32\\OpenSSH\\;C:\\cinc-project\\cinc\\bin\\;c:\\Program Files\\Zulu\\zulu-8\\bin\\"
        key_name "path"
        delim ";"
      end

      System Info:
      ------------
      chef_version=16.13.16
      platform=windows
      platform_version=10.0.17763
      ruby=ruby 2.7.3p183 (2021-04-05 revision 6847ee089d) [x64-mingw32]
      program_name=C:/cinc-project/cinc/bin/cinc-client
      executable=C:/cinc-project/cinc/bin/cinc-client


    ================================================================================
    Error executing action `add` on resource 'windows_path[C:/Program Files/PostgreSQL/13/bin/]'
    ================================================================================

    RangeError
    ----------
    windows_env[path] (C:/cinc-project/cinc/embedded/lib/ruby/gems/2.7.0/gems/chef-16.13.16-universal-mingw32/lib/chef/resource/windows_path.rb line 72) had an error: RangeError: bignum too big to convert into `long'

    Resource Declaration:
    ---------------------
    # In C:/cinc/cache/cookbooks/.../recipes/server.rb

     22: windows_path "C:/Program Files/PostgreSQL/13/bin/" do
     23:   action :add
     24: end

    Compiled Resource:
    ------------------
    # Declared in C:/cinc/cache/cookbooks/.../recipes/server.rb:22:in `from_file'

    windows_path("C:/Program Files/PostgreSQL/13/bin/") do
      action [:add]
      default_guard_interpreter :default
      declared_type :windows_path
      cookbook_name "..."
      recipe_name "server"
      path "C:/Program Files/PostgreSQL/13/bin/"
    end

    System Info:
    ------------
    chef_version=16.13.16
    platform=windows
    platform_version=10.0.17763
    ruby=ruby 2.7.3p183 (2021-04-05 revision 6847ee089d) [x64-mingw32]
    program_name=C:/cinc-project/cinc/bin/cinc-client
    executable=C:/cinc-project/cinc/bin/cinc-client

```

**Comment by ramereth** [comment]
FWIW this sounds similar to issues I've run into recently when updating various Sous Chefs cookbooks for windows. It only seems to happen on Chef 16 but not on Chef 15.

**Comment by tas50** [comment]
Fixed in #11472 which will go out in 17.1
